### ISSUE TITLE
'make check' (on master) fails 1 test on Fedora 38

### ISSUE THREAD

**Issue by mhoes** [linux-test-project/lcov#202]
Hi,

When I run 'make check' of the latest git version on a Fedora 38 install, 2 of the test that 'make check' runs fail for me (they succeed on Fedora 37). I realize that Fedora 38 is still in beta, but I'm not so sure that the fact that it's beta is the reason these tests fail. If I had to guess, I would assume (but you know what happens when we assUme) that the reason might be the higher version of gcc (v12 vs v13) ?

Anyway, these are the failing tests :

gendiffcov/simple/script.sh ....... [fail] (time 12.5s, mem 60.7MB)
gendiffcov/function/function.sh ... [fail] (time 4.7s, mem 60.8MB)

Iv'e attached the output of 'make check' and './tests/test.log' for reference, in case that helps.

[lcov-make-check.tar.gz](https://github.com/linux-test-project/lcov/files/11030791/lcov-make-check.tar.gz)

**Comment by mhoes** [comment]
Ok, the 2nd test that failed ('gendiffcov/function/function.sh') now succeeds after I installed 'xlsxwriter' packages. Sorry, I should have thought of that before.

Test 'gendiffcov/simple/script.sh' still fails though :

```
gendiffcov/simple/script.sh ....... [fail] (time 12.5s, mem 60.0MB)
    Skipping 691 previous lines (see /home/maarten/src/lcov/tests//test.log)
    ...
    Using intermediate gcov format
    Writing temporary data to /tmp/geninfo_datthOa
    Scanning . for .gcda files ...
    Found 1 data files in .
    Processing ./a-test.gcda
    geninfo: Error: 'exclude' pattern '*/iostream' is unused.
        (use "geninfo --ignore-errors unused ..." to bypass this error)
    Finished .info-file creation
    ERROR: lcov --capture failed
  EXITCODE ...: 1

```

**Comment by henry2cox** [comment]
Not sure if it will have fixed the issues or not - but I fixed some bugs and did a bit more work on portability testing (though for older gcc versions - not so much for newer).
Could you try again and see what is still broken?
If still broken:  could you run './script.sh --verbose' from the test directory and append the log (or the tail of the log).
Thanks
Henry

**Comment by mhoes** [comment]
It looks like the file `'./tests/lcov/exception/exception.sh'` does not have its permissions set to 'executable', which probably is not what was intended. ;)

After fixing that, all tests now succeed, both on Fedora 37 and Fedora 38. Thanks.

**Comment by henry2cox** [comment]
fixed in 56c0a2ab8d054db

**Comment by mhoes** [comment]
Thanks again.
